[main] Upgrade to latest dependencies

knative-automation opened the pull request:
Cron. -knative-prow-robot

/cc we don't use Prow
/assign we don't use Prow

Produced via:
```shell
./hack/update-deps.sh --upgrade --release v0.22
./hack/update-codegen.sh
```

Details:
```
bumping knative.dev/pkg 0f8d8de...c326b70:
  > c326b70 Add RunAndSyncInformers helper that makes sure informers are synced in tests (# 2055)
  > 9eeb66f Update common github actions (# 2056)
  > bbe0bb3 Implement getter machinery to generically fetch all injected clients (# 2054)
  > 85bc723 Add benchmark for metrics recording (# 2052)
  > cc278e1 upgrade to latest dependencies (# 2053)
bumping knative.dev/hack f99a255...9b73a25:
  > 9b73a25 Drop unnecessary occurrences of master (# 53)
bumping knative.dev/eventing 37e7027...9be1470:
  > 9be1470 Update common github actions (# 5053)
  > 4c5b31b upgrade to latest dependencies (# 5045)
  > ec309b1 bash v4 is required (# 5044)
  > ce0bec7 swap lberk / slinkydeveloper (# 5048)
  > 879407f smoke test more variants of pingsource (# 5042)
  > 0e2557b Update common github actions (# 5034)
  > 2fb1dd9 upgrade to latest dependencies (# 5035)
  > 2164ff0 Format markdown (# 5037)
  > 5b2912a fix # 4986 remove pingsource adapter (# 5002)
```

n3wscott reviewed (dismissed):
/lgtm

Good job bot!

knative-automation committed:
upgrade to latest dependencies

bumping knative.dev/pkg 0f8d8de...c326b70:
  > c326b70 Add RunAndSyncInformers helper that makes sure informers are synced in tests (# 2055)
  > 9eeb66f Update common github actions (# 2056)
  > bbe0bb3 Implement getter machinery to generically fetch all injected clients (# 2054)
  > 85bc723 Add benchmark for metrics recording (# 2052)
  > cc278e1 upgrade to latest dependencies (# 2053)
bumping knative.dev/hack f99a255...9b73a25:
  > 9b73a25 Drop unnecessary occurrences of master (# 53)
bumping knative.dev/eventing 37e7027...9be1470:
  > 9be1470 Update common github actions (# 5053)
  > 4c5b31b upgrade to latest dependencies (# 5045)
  > ec309b1 bash v4 is required (# 5044)
  > ce0bec7 swap lberk / slinkydeveloper (# 5048)
  > 879407f smoke test more variants of pingsource (# 5042)
  > 0e2557b Update common github actions (# 5034)
  > 2fb1dd9 upgrade to latest dependencies (# 5035)
  > 2164ff0 Format markdown (# 5037)
  > 5b2912a fix # 4986 remove pingsource adapter (# 5002)

Signed-off-by: Knative Automation <automation@knative.team>
